src/types.ts

```diff
@@ -42,26 +52,47 @@ export interface ChangeOptions {
   shouldRemoveMetadata?: boolean
 }
 
+// Events
+
+export type Events =
+  | 'healthCheckFail'
+  | 'resync'
+  | 'schemaChange'
+  | 'stateChange'
+
 interface InitialScanFailEvent {
+  type: 'healthCheckFail'
   failureType: 'initialScan'
   lastSyncedAt: number
 }
 
 interface ChangeStreamFailEvent {
+  type: 'healthCheckFail'
   failureType: 'changeStream'
-  lastRecordCreatedAt: number
+  lastRecordUpdatedAt: number
   lastSyncedAt: number
 }
 
 export type HealthCheckFailEvent = InitialScanFailEvent | ChangeStreamFailEvent
 
+export interface ResyncEvent {
+  type: 'resync'
+}
+
 export interface SchemaChangeEvent {
+  type: 'schemaChange'
   previousSchema?: JSONSchema
   currentSchema: JSONSchema
 }
 
-export type JSONSchema = Record<string, any>
+export interface StateChangeEvent {
+  type: 'stateChange'
+  from: string
+  to: string
+}
 
-export type Events = 'schemaChange' | 'healthCheckFail' | 'stateChange'
+// State
```

Comment:
State?

Reply:
The comments indicate what the types are for grouped in blocks - options, events, and state. It's just an organizational thing.